lib/streamlit/web/server/routes.py

```diff
@@ -148,13 +148,18 @@ async def get(self):
             self.write(msg)
             self.set_status(200)
 
-            # Tornado will set the _xsrf cookie automatically for the page on
+            # Tornado will set the _streamlit_xsrf cookie automatically for the page on
             # request for the document. However, if the server is reset and
             # server.enableXsrfProtection is updated, the browser does not reload the document.
             # Manually setting the cookie on /healthz since it is pinged when the
             # browser is disconnected from the server.
             if config.get_option("server.enableXsrfProtection"):
-                self.set_cookie("_xsrf", self.xsrf_token)
+                cookie_kwargs = self.settings.get("xsrf_cookie_kwargs", {})
```

Comment:
Is `xsrf_cookie_kwargs` currently defined anywhere? I couldn't find anything referencing it

Reply:
Not yet, but it is used by default by [Tornado itself](https://www.tornadoweb.org/en/stable/web.html#tornado.web.RequestHandler.xsrf_token) when t[ornado itself sets `_xsrf` cookie](https://github.com/tornadoweb/tornado/blob/b3f2a4bb6fb55f6b1b1e890cdd6332665cfe4a75/tornado/web.py#L1509), and in order to keep it consistent, it is better to add this `cookie_kwargs` when we set the cookie manually ourselves. 

We hope to use it to set `Secure=True` and `SameSite=None` (to resolve https://github.com/streamlit/streamlit/issues/5793) but right now we have problem with Safari for that